game-headed/src/main/java/org/triplea/game/client/HeadedGameRunner.java

```diff
@@ -12,6 +40,48 @@ private HeadedGameRunner() {}
    * Entry point for running a new headed game client.
    */
   public static void main(final String[] args) {
-    GameRunner.start(args);
+    checkNotNull(args);
+    checkState(!GraphicsEnvironment.isHeadless(),
+        "UI client launcher invoked from headless environment. This is currently prohibited by design to "
+            + "avoid UI rendering errors in the headless environment.");
+    ClientSetting.initialize();
+    if (!ClientSetting.useExperimentalJavaFxUi.getValueOrThrow()) {
+      Interruptibles.await(() -> SwingAction.invokeAndWait(() -> {
+        LookAndFeel.initialize();
+        initializeLogManager(Console.newInstance());
+        ErrorMessage.enable();
+      }));
+    }
+    ArgParser.handleCommandLineArgs(args);
+
+    if (SystemProperties.isMac()) {
+      MacOsIntegration.addOpenUriHandler(uri -> {
+        final String encoding = StandardCharsets.UTF_8.displayName();
+        try {
+          final String mapName = URLDecoder.decode(
+              uri.toString().substring(ArgParser.TRIPLEA_PROTOCOL.length()), encoding);
+          SwingUtilities.invokeLater(() -> DownloadMapsWindow.showDownloadMapsWindowAndDownload(mapName));
+        } catch (final UnsupportedEncodingException e) {
+          throw new AssertionError(encoding + " is not a supported encoding!", e);
+        }
+      });
+    }
+
+    if (HttpProxy.isUsingSystemProxy()) {
+      HttpProxy.updateSystemProxy();
+    }
+
+    if (ClientSetting.useExperimentalJavaFxUi.getValueOrThrow()) {
+      Services.loadAny(JavaFxClientRunner.class).start(args);
```

Comment:
Just noting that the `ServiceLoader` indirection used to start the JFX client is probably no longer necessary now that this line and the JFX client itself are once again in the same module.